src/Secretin.js

```diff
@@ -126,16 +126,23 @@ class Secretin {
                 );
               }));
         } else if (cacheAction.action === 'editSecret') {
-          return promise.then(() =>
-            decryptRSAOAEP(cacheAction.args[2], this.currentUser.privateKey)
-              .then(metadatas => {
+          return promise.then(() => {
+            let content;
+            let metadatas;
+            return decryptRSAOAEP(cacheAction.args[2], this.currentUser.privateKey)
+              .then(rawMetadatas => {
+                metadatas = rawMetadatas;
                 this.currentUser.metadatas[cacheAction.args[0]] = metadatas;
                 return decryptRSAOAEP(
                   cacheAction.args[1],
                   this.currentUser.privateKey
                 );
               })
-              .then(content => this.editSecret(cacheAction.args[0], content))
+              .then(rawContent => {
+                content = rawContent;
+                return this.editSecret(cacheAction.args[0], content)
+              })
+              .then(() => {}, () => this.addSecret(`${metadatas.title} (Conflict)`, content))
```

Comment:
It's a very rare specific case so now it's handled but we should think how to handle it better later